src/sinks/aws_kinesis_streams/integration_tests.rs

```diff
@@ -27,16 +27,16 @@ async fn kinesis_put_records() {
 
     ensure_stream(region.clone(), stream.clone()).await;
 
+    let mut batch = BatchConfig::default();
+    batch.max_events = Some(2);
+
```

Comment:
So, because the type marker fields are private, `BatchConfig` can't be constructed with the struct literal syntax.

We could have made them public but it felt like it would have been a gross approach.  I was considering another approach that adds a constructor that's only exposed in tests in something, to set all the primary fields... maybe that'd be worth it to clean this up?

Could also be a follow-up item.

Reply:
Ah, that's right. Not a big deal.